src/modules/Packages.rb

```diff
@@ -1129,118 +1079,52 @@ def ComputeSystemPatternList
-
-      if Ops.greater_than(Builtins.size(@additional_packages), 0)
-        Builtins.y2warning(
-          "Additional packages are still in use, please, change it to use PackagesProposal API"
-        )
-        Builtins.y2milestone("Additional packages: %1", @additional_packages)
-        install_list = Convert.convert(
-          Builtins.union(install_list, @additional_packages),
-          :from => "list",
-          :to   => "list <string>"
-        )
+      if !@additional_packages.empty?
+        log.warn("Additional packages are still in use, please, change it to use PackagesProposal API")
+        log.info("Additional packages: #{@additional_packages}")
+        install_list.concat(@additional_packages)
       end
 
       # bnc #431580
       # New API for packages selected by other modules
-      packages_proposal_all_packages = PackagesProposal.GetAllResolvables(
-        :package
-      )
-      if Ops.greater_than(Builtins.size(packages_proposal_all_packages), 0)
-        Builtins.y2milestone(
-          "PackagesProposal::GetAllResolvables returned: %1",
-          packages_proposal_all_packages
-        )
-        install_list = Convert.convert(
-          Builtins.union(install_list, packages_proposal_all_packages),
-          :from => "list",
-          :to   => "list <string>"
-        )
-      else
-        Builtins.y2milestone("No packages required by PackagesProposal")
-      end
+      # use both mandatory and optional packages
+      packages_proposal_all_packages = PackagesProposal.GetAllResolvables(:package, optional: true)
+      packages_proposal_all_packages.concat(PackagesProposal.GetAllResolvables(:package))
+
+      log.info("PackagesProposal packages: #{packages_proposal_all_packages}")
+      install_list.concat(packages_proposal_all_packages)
 
       # Kernel is added in autoinstPackages () if autoinst is enabled
       if !Mode.update || !Mode.autoinst
         kernel_pkgs = Kernel.ComputePackages
-        kernel_pkgs_additional = ComputeAdditionalKernelPackages()
-        install_list = Convert.convert(
-          Builtins.union(install_list, kernel_pkgs),
-          :from => "list",
-          :to   => "list <string>"
-        )
-        if Ops.greater_than(Builtins.size(kernel_pkgs_additional), 0) &&
-            kernel_pkgs_additional != nil
-          install_list = Convert.convert(
-            Builtins.union(install_list, kernel_pkgs_additional),
-            :from => "list",
-            :to   => "list <string>"
-          )
-        end
-      end
+        install_list.concat(kernel_pkgs)
 
-      if Pkg.IsSelected("xorg-x11-Xvnc") && Linuxrc.vnc
-        install_list = Convert.convert(
-          Builtins.union(install_list, graphicPackages),
-          :from => "list",
-          :to   => "list <string>"
-        )
-      else
-        Builtins.y2milestone("Not selecting graphic packages")
+        kernel_pkgs_additional = ComputeAdditionalKernelPackages()
```

Comment:
just idea, why not have in `ComputeAdditionalKernelPackages()` switch to detect mode and also return also `Kernel.ComputePackages` included? this way this can be simple one liner.